core/src/test/scala/org/apache/spark/sql/test/generator/ColumnValueGenerator.scala

```diff
@@ -245,8 +245,15 @@ case class ColumnValueGenerator(dataType: ReflectedDataType,
           randomValue(r)
         }.toList ++ specialBound
       }
+
+      val expectedGeneratedRandomValuesLen = if (generateUnique) {
+        generatedRandomValues.toSet.size
+      } else {
+        generatedRandomValues.size
+      }
+
       assert(
-        generatedRandomValues.toSet.size >= n,
+        expectedGeneratedRandomValuesLen >= n,
```

Comment:
I think it is okay to remove `toSet` and use the original way to assert.

Reply:
In this way, it can help us better locate the problem. In particular, when we are generating unique value.